sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCScanBuilder.scala

```diff
@@ -131,7 +131,8 @@ case class JDBCScanBuilder(
   }
 
   override def pushLimit(limit: Int): Boolean = {
-    if (jdbcOptions.pushDownLimit) {
+    if (jdbcOptions.pushDownLimit &&
+      JdbcDialects.get(jdbcOptions.url).canPushDownLimit(jdbcOptions, limit)) {
```

Comment:
It will still throw an out-of-memory exception anyway if 1. the limit is very large, or 2. no limit is pushed down (as it will collect all anyway). How does this option solve the problem?

Reply:
This option just used to solve out-of-memory when `Limit` could be pushed down. Because all the data will pulled by one Task when pushing down `Limit`. If Limit cannot be pushed down, Spark could uses jdbc partition or other method to avoid the problem.